addons/hr_expense/models/hr_expense.py

```diff
@@ -542,6 +542,8 @@ def _check_payment_mode(self):
 
     def _prepare_base_line_for_taxes_computation(self, **kwargs):
         self.ensure_one()
+        if 'price_unit' in kwargs:
+            kwargs['price_unit'] = kwargs.get('price_unit') or self.total_amount
```

Comment:
My problem here is that this is a overly simplistic approach to a bigger issue, and won't work/will create a lot of problems in most cases